Tensile/Source/client/main.cpp

```diff
@@ -537,11 +547,38 @@ int main(int argc, const char* argv[])
                             for(int i = 0; i < warmupInvocations; i++)
                             {
                                 listeners.preWarmup();
-                                if(gpuTimer)
-                                    adapter.launchKernels(
-                                        kernels, stream, warmupStartEvents[i], warmupStopEvents[i]);
+
+                                if(i == 1 && boundsCheck == BoundsCheckMode::GuardPageAll)
```

Comment:
This breaks the encapsulation of the DataInitialization class.  If you need to send `i` into the prepareGPUInputs function, then do that.

Reply:
@sdquiring If we put i into prepareGPU, every kernel launch need to re-call prepareGPUInputs() and solve() one time. launch index "i" is meaningless for DataInitialization class.
For "GuardPageAll", it needs construct 2 sets DataInitialization. Different "launch kernel" needs to call different dataInit. It would break encapsulation of the DataInitialization class no matter we put the implementation inside or outside of DataInitialization class.
I would like to discard my third commit "GuardPageAll" and just keep the implementation "GuardPageFront" and "GuardPageBack".
Is that ok?

Reply:
If we can validate front and back in one run, this feature is much more likely to be used.

What if this mode caused there to be two iterations of the `while(listeners.needMoreRunsInSolution())` loop?  This could be controlled from the ReferenceValidator, or we could make the DataInitialization class be a listener.